textFilter.js

```diff
@@ -0,0 +1,63 @@
+// selects the elements in DOM
+const searchInput = document.querySelector('.input-search-keyword');
+const roomsUl = document.querySelector('.rooms');
+
+function search(){
+    const preSearchString = searchInput.value;
+    //if string empty display all rooms
+    if(preSearchString.lenght <= 0){
+        toRenderAll;
+    }
+    //if string is not empty render rooms matching the string
+    else{
+        filter(preSearchString);
+    }
+}
+
+// renders all rooms if the input (searchString) is empty
+async function toRenderAll(){
+    const data = await getChallenges();
+    roomsUl.innerText = '';
+
+    data.challenges.forEach(challenge => {
+        render(challenge.id, challenge.title, challenge.description);
+    });
+}
+
+// renders all the rooms that matches their title and or description with searchString.
+function render(id, title, description){
+    const item = document.createElement("li");
+    const content = document.createElement("p");
+    content.innerText = id + "\nTitle: " + title + "\nDescription: " + description;
+    item.appendChild(content);
+    roomsUl.appendChild(item);
+}
+
+async function filter(preSearchString){
+    const data = await getChallenges();
+    // makes the preSearchString to lowercase so the search ignores if you use upper or lower case
+    const searchString = preSearchString.toLowerCase();
+    roomsUl.innerText = ''; 
+    data.challenges.forEach(challenge => {
+        // makes the title and description to lowercase to make the search not caseSensitive LCTitle/LCDescription == LowerCaseTitle/Description 
+        const LCTitle = challenge.title.toLowerCase();
```

Comment:
This naming convention, with the capital first letter of the word (`LCTitle` instead of `lcTitle` or something like that) is unusual for variables in javascript. Most developers and teams have as a convention to use `camelCase` for variables and `PascalCase` (what you're using here) for classes only.

If you want to use a naming convention like this, it's important to be consistent. I was confused to see these variables, because I first thought they were something other than regular variables, since it's such an uncommon naming convention and not consistent with other variables in this project.